be/src/storage/lake/async_delta_writer.cpp

```diff
@@ -194,7 +197,7 @@ inline void AsyncDeltaWriterImpl::finish(Callback cb) {
 
 inline void AsyncDeltaWriterImpl::close() {
     std::lock_guard l(_mtx);
-    if (_opened && !_closed) {
+    if (_queue_id != kInvalidQueueId) {
```

Comment:
If you change code like this, is _closed still useful

Reply:
`_closed` is still useful for preventing re`open()` the AsyncDeltaWriter after `close()`ed:
```
inline Status AsyncDeltaWriterImpl::open() {
    std::lock_guard l(_mtx);
    if (_closed) { // <----- here
        return Status::InternalError("AsyncDeltaWriter has been closed");
    }
    if (_opened) {
        return _status;
    }
    _status = do_open();
    _opened = true;
    return _status;
}
````